A80-pid.md

```diff
@@ -0,0 +1,290 @@
+  // Default is 1.
+  google.protobuf.FloatValue derivative_gain = 4;
+
+  // Maximum allowable weight. Weights proposed by the PID controller exceeding this value will be capped.
+  // This prevents infinite weight growth, which could occur if only a subset of clients uses PID and increasing weights
+  // no longer effectively corrects the imbalance. Default is 10.
+  google.protobuf.FloatValue max_weight = 5;
+
+  // Minimum allowable weight. Weights proposed by the PID controller falling below this value will be capped.
+  // This prevents weights from dropping to zero, which could occur if only a subset of clients uses PID and decreasing weights
+  // no longer effectively corrects the imbalance. Default is 0.1.
+  google.protobuf.FloatValue min_weight = 6;
+}
+```
+
+### PID controller
+
+A PID controller is a control loop feedback mechanism that continuously calculates an error value as the difference between a desired setpoint (`referenceSignal`) and a measured process variable (`actualSignal`). It then applies a correction based on proportional, integral, and derivative terms (denoted P, I, and D respectively), hence the name.
+
+In our implementation, we will not be using the integral part. The integral component is useful for speeding up convergence when the `referenceSignal` changes sharply. In our case, we will be converging the load on the subchannels to a mean value, which is mostly stable.
+
+Here is a sample implementation in pseudo-code:
+
+```
+pidController class {
+  proportionalGain float
+  derivativeGain float
+
+  controlError float
+
+  update(referenceSignal float, actualSignal float, samplingInterval duration) float {
+    previousError = this.controlError
+    // Save last controlError so we can use it to calculate derivative during next update
+    this.controlError = referenceSignal - actualSignal
+    controlErrorDerivative = (this.controlError - previousError) / samplingInterval.Seconds()
+    controlSignal = this.proportionalGain * this.controlError +
+                    this.derivativeGain * controlErrorDerivative
+    return controlSignal
+  }
+}
+```
+
+The `update` method is expected to be called on a regular basis, with `samplingInterval` being the duration since the last update. The return value is the control signal which, if applied to the system, should minimize the control error. In the next section, we'll discuss how this control signal is converted to `wrr` weight.
+
+The `proportionalGain` and `derivativeGain` parameters are taken from the LB config. `proportionalGain` should be additionally scaled by the `WeightUpdatePeriod` value. This is necessary because derivative error is calculated like `controlErrorDerivative = (this.controlError - previousError) / samplingInterval.Seconds()` and dividing by a very small `samplingInterval` value makes the result too big. `WeightUpdatePeriod` is roughly equal to `samplingInterval` as we will be updating the PID state once per `WeightUpdatePeriod`.
```

Comment:
The proposal was to scale only proportionalGain, not derivativeGain. Also I'll update the doc to explicitly mention multiplication instead of scaling.